### ISSUE TITLE
[switch] Focus ring is missing in Firefox after pointer focus and keyboard activation

### ISSUE THREAD

**Issue by lyzno1** [mui/base-ui#5130]
Browser: Firefox

https://github.com/user-attachments/assets/1c37e47d-7751-47f1-a9fc-e025272ddc3f

_Originally posted by @lyzno1 in https://github.com/mui/base-ui/issues/5129#issuecomment-4807066701_

**Comment by mj12albert** [comment]
This is just Firefox's behavior and not an issue in the library, it occurs with a plain input/button elements: https://stackblitz.com/edit/stackblitz-starters-3awbpfhj?file=index.html

**Comment by lyzno1** [comment]
Thanks for clarifying. Since this also reproduces with a plain button in Firefox, I'll close this as expected browser behavior.
